packages/x-data-grid/src/hooks/features/listView/useGridListView.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react';
+import type { GridStateColDef } from '../../../models/colDef/gridColDef';
+import { GridStateInitializer } from '../../utils/useGridInitializeState';
+import { useGridApiMethod } from '../../utils/useGridApiMethod';
+import { GridPrivateApiCommunity } from '../../../models/api/gridApiCommunity';
+import { GridListViewApi } from '../../../models/api/gridListViewApi';
+import { DataGridProcessedProps } from '../../../models/props/DataGridProps';
+import { GridEventListener } from '../../../models/events';
+import {
+  gridListColumnSelector,
+  gridVisibleListColumnDefinitionsSelector,
+} from './gridListViewSelectors';
+import { gridColumnDefinitionsSelector, gridVisibleColumnDefinitionsSelector } from '../columns';
+import { gridDimensionsSelector } from '../dimensions';
+import { useGridApiEventHandler } from '../../utils/useGridApiEventHandler';
+
+export type GridListViewState = { listColumn: GridStateColDef | undefined };
+
+export const listViewStateInitializer: GridStateInitializer<
+  Pick<DataGridProcessedProps, 'unstable_listColumn'>
+> = (state, props, apiRef) => ({
+  ...state,
+  listView: {
+    listColumn: { ...props.unstable_listColumn, computedWidth: getListColumnWidth(apiRef) },
+  },
+});
+
+export function useGridListView(
+  apiRef: React.MutableRefObject<GridPrivateApiCommunity>,
+  props: Pick<DataGridProcessedProps, 'unstable_listColumn'>,
+) {
+  /*
+   * API METHODS
+   */
+  const getListColumn: GridListViewApi['getListColumn'] = (field) => {
+    const listColumn = gridListColumnSelector(apiRef.current.state);
+    if (listColumn?.field === field) {
+      return listColumn;
+    }
+
+    const columns = gridColumnDefinitionsSelector(apiRef);
+    return columns.find((col) => col.field === field && col.type === 'actions');
+  };
+
+  const getListColumnIndex: GridListViewApi['getListColumnIndex'] = (field) => {
+    const columns = gridVisibleListColumnDefinitionsSelector(apiRef);
+    return columns.findIndex((col) => col.field === field);
+  };
+
+  const listColumnApi: GridListViewApi = {
+    getListColumn,
+    getListColumnIndex,
+  };
+
+  useGridApiMethod(apiRef, listColumnApi, 'private');
+
+  /*
+   * EVENTS
+   */
+  const updateListColumnWidth = () => {
+    apiRef.current.setState((state) => {
+      return {
+        ...state,
+        listView: {
+          ...state.listView,
+          listColumn: {
+            ...state.listView.listColumn,
+            computedWidth: getListColumnWidth(apiRef),
+          },
+        } as GridListViewState,
```

Comment:
I think we can get rid of these, I can give it a try

Reply:
Done in [`1d23f2f` (#14393)](https://github.com/mui/mui-x/pull/14393/commits/1d23f2f39747e85152e3bfd1d2c162748c377bc0)